client/src/main/scala/com/redi/scalademo/presentation/FormListener.scala

```diff
@@ -0,0 +1,105 @@
+package com.redi.scalademo.presentation
+
+import autowire.{clientCallable, unwrapClientProxy}
+import com.redi.scalademo.business.{Calculator, NumericStringValidator, ValidatedResult}
+import com.redi.scalademo.infrastructure.Client
+import org.scalajs.dom
+import org.scalajs.jquery.{JQuery, JQueryEventObject, jQuery ⇒ $}
+
+import scala.concurrent.Future
+import scala.math.BigDecimal
+import scala.scalajs.concurrent.JSExecutionContext.Implicits.queue
+import scala.scalajs.js
+
+class FormListener(
+  client: Client,
+  numericStringValidator: NumericStringValidator
+) {
+
+  def attachTo(formId: String): Unit = {
+
+    val InfoClass = "is-info"
+    val DangerClass = "is-danger"
+
+    val Validate = "validate"
+    val Augend = "augend"
+    val Addend = "addend"
+    val Summand = "summand"
+
+    val Name = "name"
+    val Value = "value"
+
+    val formElement: JQuery = $(s"form#$formId")
+    val summandElement: JQuery = formElement.find(elementNamed(Summand))
+
+    formElement.submit { (e: JQueryEventObject) ⇒
+
+      e.preventDefault()
+
+      formElement.find(namedElement).removeClass(DangerClass)
+      val serializedFormData = formElement.serializeArray().asInstanceOf[js.Array[js.Dictionary[String]]]
+      val (validationFormData: js.Array[js.Dictionary[String]], calculableFormData: js.Array[js.Dictionary[String]]) =
+        serializedFormData.partition(_(Name) == Validate)
+
+      val shouldValidate: Boolean = validationFormData.forall(_(Value).nonEmpty)
+
+      var allFormValuesAreValid = true
+      var augend: String = ""
+      var addend: String = ""
+
+      for (formControlData: js.Dictionary[String] ← calculableFormData) {
```

Comment:
Perhaps I'll move the appropriate variables to the the class instance rather than being local.